### PR TITLE
Focus indicators consistent

### PR DESCRIPTION
**Pull request by abdul99ahad**
### Tab action buttons 

Earlier: no focus 
Now: 
image

### Checkbox
Earlier: 
image

Now: 
image

### Buttons 
image

Changes: 
image

### Bottom panel tab bar 

Earlier: no focus 
Changes: 
image (changes required)

### Bottom panel icons 
Earlier: no focus 
Changes: 
image

### Bottom panel focus alignment issue 

Earlier: image
Changes: 
image

### Overlay buttons 
Earlier: image

Changes: 
image

### Footer icons
Earlier: grey background 
image

Changes: 
image

### Variable outline tab 
Earlier: no focus 
Changes: 
image

Closes https://github.com/camunda/camunda-modeler/issues/4632

Depends on https://github.com/bpmn-io/variable-outline/pull/6
Depends on https://github.com/bpmn-io/properties-panel/pull/390

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] __Brief textual description__ of the changes present
* [x] __Visual demo__ attached
* [ ] __Steps to try out__ present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

### CODE REVIEW

File: client/src/app/panel/tabs/variable-outline/VariableOutlineTab.less [hunk 1]
```diff
@@ -24,6 +24,18 @@
       font-size: 14px !important;
     }
   }
+  
+  .bio-vo-text-button {
```
Comment (nikku) [inline client/src/app/panel/tabs/variable-outline/VariableOutlineTab.less:28]: Is this about the specific variable outline or a general Carbon customization?

File: client/src/app/resizable-container/PropertiesPanelContainer.less [hunk 2]
```diff
@@ -21,4 +21,15 @@
     --font-family-monospace: inherit;
     background: var(--color-white);
   }
+
+  .bio-properties-panel-input[type="checkbox"]:focus {
```
Comment (nikku) [inline client/src/app/resizable-container/PropertiesPanelContainer.less:25]: Should this be a general properties panel fix?

**Review by nikku — COMMENTED** [review]
Good progress, some early comments:

* Cannot be focused yet: 
  image

* I see two versions of the outline (thin, and thick), do we plan to align this?
  image
  image

* I see two versions of outline styles now for buttons, do we plan to align this?
  image
  image

* Collapsed properties panel is still part of focusable (tabbable) application; we should `detach` it once collapsed:
  
  capture 4H6aN7_optimized


Generally I'd suggest we fix the outlines in the relevant libraries, where they are broken, and don't monkey patch in modeler. Applies to properties panel and variable outline.

File: client/src/app/panel/tabs/variable-outline/VariableOutlineTab.less (hunk 1 above)
Comment (abdul99ahad) [reply client/src/app/panel/tabs/variable-outline/VariableOutlineTab.less:28]: I don't think this is general carbon customization. Nevertheless, we have to see in `variable-outline` project.

**Comment by abdul99ahad** [comment]
image

Improved the tab bar icon focus. The styles are consistent and I think the difference is very minor if you look closely. Do we invest more time to align this more perfectly or is it good to have? 

cc: @nikku @lmbateman

**Comment by abdul99ahad** [comment]
> image
> Improved the tab bar icon focus. The styles are consistent and I think the difference is very minor if you look closely. Do we invest more time to align this more perfectly or is it good to have?
> 
> cc: @nikku @lmbateman

I think it's much better now. 
image

**Comment by abdul99ahad** [comment]
Summary of the changes: 
- Tabs and tab action buttons have focus 
- Collapsed panel elements won't get focus 
- Bottom bar have same consistent focus states 
- Adjusted focus states offset for certain elements (buttons, containers, etc.,) 
- Properties panel focus states are fixed [here](https://github.com/bpmn-io/properties-panel/pull/390)
- Variable outline focus states are fixed [here](https://github.com/bpmn-io/variable-outline/pull/6)

**Comment by lmbateman** [comment]
> Collapsed properties panel is still part of focusable (tabbable) application; we should detach it once collapsed (Nico, above)
> resolved collapsed panel to be not part of focus anymore (Abdul, in Slack)

I'm a little confused. Does this mean that
- when the Properties panel is collapsed, it completely drops out of the tab sequence, or
- when the Properties panel is collapsed, focus lands once on the panel as a whole, but not the items inside the panel?

**Comment by nikku** [comment]
@abdul99ahad reviewed + merged both upstream PRs:

* https://github.com/bpmn-io/variable-outline/pull/6
* https://github.com/bpmn-io/properties-panel/pull/390

File: resources/plugins/test/style.css [hunk 3]
```diff
@@ -1,7 +1,3 @@
-.bjs-powered-by {
```
Comment (nikku) [inline resources/plugins/test/style.css:1]: Why this change?

Comment (abdul99ahad) [reply resources/plugins/test/style.css:1]: The bpmn.io watermark was removed in this [commit](https://github.com/camunda/camunda-modeler/commit/b1102b4e43db10aa56923816b929f826345c1c67). However, this `block` forces the watermark to remain visible in the modeler and allows it to gain focus when navigating via keyboard tabbing.
Screenshot 2024-12-09 at 14 52 51

**Comment by lmbateman** [comment]
In general, this is a huge improvement! Thanks for all your hard work.

I turned up a handful of additional issues, including a couple I missed on the first round (sorry). I'm not sure if it makes sense to fix these as part of this issue, or to open another issue. What's the standard practice? Also, I think a couple of these are judgment calls, so I'd be interested in hearing any thoughts from the team.

1. What does the focus indicator on the tab do? Maybe we should only have focus indicators on the tab Close button (unless this has to do with keyboard binding).
2. I think we missed the focus indicators on the + and ... dropdown menus. The menu items still only have gray backgrounds.
3. There's an extra tab stop between ... and the General section's accordion button. Is this focusing the Properties panel?
4. I'm willing to discuss this one, but I was surprised when I pressed Return on a row in the Problems panel and the focus shifted to the erroneous field in the Properties panel. I guess this boils down to two questions:
   - How easy is it to learn this behavior? (I would say fairly easy; after I've done it once or twice, I'll probably remember.)
   - How beneficial or annoying is this behavior?
7. The tooltip icon, Elements header, and Variables header in the Variables panel still doesn't get focus. I'm OK either way -- the user will probably only want to read them once -- but we should be consistent with the Properties panel, where headers with tooltips do currently get focus.
8. I noticed the table and the tree in the Variables panel behave differently; when focus lands on the tree, it lands on the root node, and then you can use the arrow keys to navigate through the tree. In the table, focus lands on every cell. This might get annoying when the table is large. https://www.w3.org/WAI/ARIA/apg/patterns/grid/examples/data-grids/ has examples of tables that are initially accessed through the Tab key but then use arrow keys for internal navigation. (To be clear, I think it's OK for focus to move through the table headers, land on the first cell, and then land on the XML button.)
9. When I pressed Return on the XML button, it showed me the XML, with focus remaining on the button; and then when I pressed Return again, it showed the diagram, with the focus on the first interactive item in the Properties panel of the selected element. I think focus should remain on the XML button in both cases.
10. It might be my eyes, but could you please check the colors on the three elements in the Camunda version dialog? It seems like the focus indicator on the dropdown field has a lighter blue than the ones on the link and the Apply button.
11. I missed the Remember credentials toggle in the Deploy dialog; sorry! It doesn't have a visible indicator.
12. When the Deploy dialog opens, focus is on the Cluster endpoint field. Does this make sense from a developer's perspective?
13. When I switch the target to Camunda 8 SaaS, focus remains in the radio button group. However, when I switch it to Camunda 8 self-managed, focus moves to the first text field. We should be consistent. (I would leave it in the radio button field.)
14. Pressing Return on the Problems list or the Variables button in the status bar toggles the bottom panel. I think this is OK.

**Comment by nikku** [comment]
> What's the standard practice? 

To follow-up on things we found as part of this initiative, or clearly capture them in issues , or agree the are won't fix.

> What does the focus indicator on the tab do? Maybe we should only have focus indicators on the tab Close button (unless this has to do with keyboard binding).

Tab can potentially be moved, it is a selectable thing, and hence I'd keep it focusable.

> I think we missed the focus indicators on the + and ... dropdown menus. The menu items still only have gray backgrounds.

Agreed.

capture Q8wALL_optimized

> There's an extra tab stop between ... and the General section's accordion button. Is this focusing the Properties panel?

It is focusing the canvas, cf. https://github.com/bpmn-io/diagram-js/pull/662, but we don't indicate the focus to the user (we should).

> I'm willing to discuss this one, but I was surprised when I pressed Return on a row in the Problems panel and the focus shifted to the erroneous field in the Properties panel. I guess this boils down to two questions:

What do you expect instead? Return or SPACE does always (unless you are in a form) trigger the link action (or click behavior). Which is to navigate to the thing. What we could do instead:

* When focusing via keyboard, show a special "navigate to problem in problems panel" link. This would only be shown when keyboard focus is on the element.

> I noticed the table and the tree in the Variables panel behave differently; when focus lands on the tree, it lands on the root node, and then you can use the arrow keys to navigate through the tree. In the table, focus lands on every cell. This might get annoying when the table is large. https://www.w3.org/WAI/ARIA/apg/patterns/grid/examples/data-grids/ has examples of tables that are initially accessed through the Tab key but then use arrow keys for internal navigation. (To be clear, I think it's OK for focus to move through the table headers, land on the first cell, and then land on the XML button.)

This is mandated by Carbon, unless I'm mistaken. I'd not invest in working around the Carbon behaviors.

> When I pressed Return on the XML button, it showed me the XML, with focus remaining on the button; and then when I pressed Return again, it showed the diagram, with the focus on the first interactive item in the Properties panel of the selected element. I think focus should remain on the XML button in both cases.

Likely not an easy fix, let's create a clearly reproducible `a11y` tagged follow-up issue.

> When the Deploy dialog opens, focus is on the Cluster endpoint field. Does this make sense from a developer's perspective?

Only if it is empty, I think, otherwise the canoncal action is to "just press enter" to confirm config and deploy.

However I'd not invest there right now. We're planning to rebuild this anyway shortly.

> Pressing Return on the Problems list or the Variables button in the status bar toggles the bottom panel. I think this is OK.

Could you share a recording? I'm not sure I can reproduce it.

**Review by nikku — APPROVED** [review]
This works well, and while there [is still things to tackle as a potential follow-up](https://github.com/camunda/camunda-modeler/pull/4728#issuecomment-2537345986) I see the current state as _worth releasing._

**Comment by lmbateman** [comment]
> This works well, and while there https://github.com/camunda/camunda-modeler/pull/4728#issuecomment-2537345986 I see the current state as worth releasing.
Agreed; it's a HUGE improvement already.
